fastify.js

```diff
@@ -631,7 +631,7 @@ function fastify (options) {
   function setSchemaController (schemaControllerOpts) {
     throwIfAlreadyStarted('Cannot call "setSchemaController" when fastify instance is already started!')
     const old = this[kSchemaController]
-    const parent = old.parent || old
+    const parent = old
```

Comment:
If the issue is the plugin registration with a null ref, shouldn't this be smth like this?
```suggestion
    const parent = old ? old.parent : old
```

Reply:
Not exactly, the issue appears on every new registration of a `SchemaController`, if you register a new one within a new encapsulated context, as the current context where the controller is being registered as a parent, any added schema within the current context is missed, causing to (in this case Ajv) totally missed such registered schemas.

By following the suggestion, see the result:
![image](https://user-images.githubusercontent.com/10200776/142506181-110d6bb0-8703-44df-bb71-a173bd3571f7.png)

Having in mind that:
- `some` was registered in the root instance
- `another` was registered in the child instance

Reply:
Added a test without a plugin for better explanation 🙂 
4877f81

Reply:
I would remove the `parent` variable since it is no more useful.

To help understand what is going on here, be aware that the `compilersFactory` is used only when necessary.

In this case, the root context does not use any schemas because it does not contain any route.
For this reason, the "main" compilers is not built.

As you can see, the plugin context instead uses a route and try to build the compilers, triggering the error (`old` is null)

Reply:
Removed in 79213ce 👍